src/XmlPayload.php

```diff
@@ -0,0 +1,30 @@
+<?php
+declare(strict_types = 1);
+
+namespace Middlewares;
+
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use SimpleXMLElement;
+
+class XmlPayload extends Payload implements MiddlewareInterface
+{
+    /**
+     * @var array
+     */
+    protected $contentType = ['text/xml', 'application/xml', 'application/x-xml'];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function parse(StreamInterface $stream)
+    {
+        $string = trim((string)$stream);
+
+        if (empty($string)) {
```

Comment:
`empty()` returns `true` for non-empty values like `"0"`, so it's better to check `if ($string === '')`

Reply:
Yep, you're right! Just changed it